lib/mail/encodings/transfer_encoding.rb

```diff
@@ -23,30 +23,39 @@ def self.cost(str)
         raise "Unimplemented"
       end
 
+      def self.should_transport?(str)
+        true
+      end
+
       def self.to_s
         self::NAME
       end
 
       def self.get_best_compatible(source_encoding, str)
-        if self.can_transport? source_encoding then
-            source_encoding 
+        if self.can_transport?(source_encoding) && self.should_transport?(str)
+          source_encoding
         else
-            choices = []
-            Encodings.get_all.each do |enc|
-                choices << enc if self.can_transport? enc and enc.can_encode? source_encoding
-            end
-            best = nil 
-            best_cost = 100
-            choices.each do |enc|
-                this_cost = enc.cost str
-                if this_cost < best_cost then
-                    best_cost = this_cost
-                    best = enc
-                elsif this_cost == best_cost then 
-                    best = enc if enc::PRIORITY < best::PRIORITY
-                end
+          choices = Encodings.get_all.select do |enc|
+            self.can_transport?(enc) && enc.can_encode?(source_encoding)
+          end
+
+          best = nil
+          best_cost = nil
+
+          choices.each do |enc|
+            # If the current choice cannot be transported safely,
+            # give priority to other choices but allow it to be used as a fallback.
+            this_cost = enc.cost(str) if enc.should_transport?(str)
```

Comment:
That would work if we want to disable the encoding completely, but isn't the same as what the comment says (infinitely low priority). Maybe disabling would be better though?